inc/blocks/namespace.php

```diff
@@ -93,6 +95,47 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 		return;
 	}
 
+	update_or_create_xb_post( $xbs, $post->post_title, $post_ID, null, 'post' );
+
+}
+
+/**
+ * Synchronise any XBs on the widget page with a shadow post type.
+ *
+ * @param array $instance The array of current widget settings.
+ * @param array $new_instance The array of new widget settings.
+ * @param array $old_instance The array of old widget settings.
+ * @param WP_Widget $widget The current widget instance.
+ * @return array $instance The current widget settings.
+ */
+function on_widgets_save( $instance, $new_instance, $old_instance, $widget ) {
+
+	if ( ! isset( $instance['content'] ) ) {
+		return $instance;
+	}
+
+	// Scan for XBs in the post content.
+	$blocks = parse_blocks( $instance['content'] );
+	$xbs = find_xbs( $blocks );
+
+	// Update / Create XB's if the content contains XB's.
+	if ( ! empty( $xbs ) ) {
+		update_or_create_xb_post( $xbs, __( 'Widget' ), null, $widget->number, 'widget' );
+	}
+
+	return $instance;
+}
+
+/**
+ * Update or create new XB shadow posts.
+ *
+ * @param array $xbs XBs on page or as widgets.
+ * @param string $default_title Default Title title for XB Post.
+ * @param int $post_ID Post ID if applicable in use case.
+ * @param int $item_id Item ID if applicable in use case.
+ * @param string $context Meta Key for updating the post_meta.
+ */
+function update_or_create_xb_post( array $xbs, string $default_title = '', ?int $post_ID = null, ?int $item_id = null, string $context = 'post' ) {
```

Comment:
@roborourke This Im not sure if you maybe have a suggestion to improve it better. 

The `$post_ID` variable which is used for the `post_parent` when doing the `wp_insert_post` and `wp_update_post`.

Then the `item_id` variable is used for when the XBs are being used in widgets. It's then added to the `$xb_title` value to differentiate the different titles.

Not sure if I am being a bit to pedantic on this change?

Reply:
I think this makes sense, it's not going to get loads of additional use and it's internal so it can always be iterated on but this approach is specific about what the arguments are in a good way without overloading anything too much. I like it.